app/controllers/submissions_controller.rb

```diff
@@ -278,15 +316,38 @@ def browse
       @grace_credit_column = ''
     end
 
-    if @assignment.past_collection_date?
-      notice_text = t('browse_submissions.grading_can_begin')
+    if @assignment.past_all_collection_dates?
+      flash_now(:notice, t('browse_submissions.grading_can_begin'))
     else
-      collection_time = @assignment.submission_rule.calculate_collection_time
-      notice_text = t('browse_submissions.grading_can_begin_after',
-                      time: I18n.l(collection_time, format: :long_date))
+      if @assignment.section_due_dates_type
+        section_due_dates = Hash.new
+        now = Time.zone.now
+        Section.all.each do |section|
+          collection_time = @assignment.submission_rule
+                                       .calculate_collection_time(section)
+          collection_time = now if now >= collection_time
+          if section_due_dates[collection_time].nil?
+            section_due_dates[collection_time] = Array.new
+          end
+          section_due_dates[collection_time].push(section.name)
+        end
+        section_due_dates.each do |collection_time, sections|
+          plural = sections.length > 1 ? 's' : '';
+          after = collection_time != now ? '_after' : ''
+          sections = sections.join(', ').reverse
+                             .sub(' ,', t('and').reverse)
+                             .reverse
+          flash_now(:notice, t('browse_submissions.grading_can_begin' +
```

Comment:
Don't dynamically generate internationalization keys. Makes it harder to search for.